### DISCUSSION TITLE
PnP core SDK code.. Error "Property PermissionRead does not exist in this entity"

### DISCUSSION THREAD

**Discussion by mvcsharepointdev** [pnp/pnpcore#1107] [Q&A] [answered]
I have the following PnP core SDK code to get a field named "PermissionRead" which is of type People or Group :-

```
try
            {
                using (var context = await pnpContextFactory.CreateAsync("Default"))
                {

                    /////////////////////////
                    // Assume the fields where not yet loaded, so loading them with the list
                    var myList = context.Web.Lists.GetByTitle("tickets", p => p.Title,
                                                                         p => p.Fields.QueryProperties(p => p.InternalName,
                                                                                                       p => p.FieldTypeKind,
                                                                                                       p => p.TypeAsString,
                                                                                                       p => p.Title));
                    // Do a paged retrieval (non async sample) of the list items with additional collection loads
                    foreach (var listItem in myList.Items.QueryProperties(
                                                         p => p.RoleAssignments.QueryProperties(p => p.PrincipalId,
                                                                                               p => p.RoleDefinitions)))
                    {

                        var t = listItem["PermissionRead"];
                      
                    }
```

but i am getting the following error:-

`    Property PermissionRead does not exist in this entity`

so how i can load this custom column?

**Comment by jansenbe** [comment, marked as answer]
@mvcsharepointdev : I assume `PermissionRead` is the internal name of the field you're loading? Do you see the other custom fields in the response?

**Reply by mvcsharepointdev** [reply]
@jansenbe yes the "PermissionRead" is the internal name of the field.. but how i can load it?

**Reply by mvcsharepointdev** [reply]
@jansenbe  the question is how we can load custom columns other than the ones available for us 

```
.QueryProperties(p => p.InternalName,
                                                                                                       p => p.FieldTypeKind,
                                                                                                       p => p.TypeAsString,
                                                                                                       p => p.Title));
```

**Reply by mvcsharepointdev** [reply]
@jansenbe yes other fields are loaded without issue, but this is a people or group field which allow multiple selection

**Reply by jansenbe** [reply]
@mvcsharepointdev : you also need to add `p => p.All` as otherwise you're only loading the role assignments. I'll update the docs to make that clear.

```csharp
/////////////////////////
                    // Assume the fields where not yet loaded, so loading them with the list
                    var myList = context.Web.Lists.GetByTitle("tickets", p => p.Title,
                                                                         p => p.Fields.QueryProperties(p => p.InternalName,
                                                                                                       p => p.FieldTypeKind,
                                                                                                       p => p.TypeAsString,
                                                                                                       p => p.Title));
                    // Do a paged retrieval (non async sample) of the list items with additional collection loads
                    foreach (var listItem in myList.Items.QueryProperties(p => p.All,
                                                         p => p.RoleAssignments.QueryProperties(p => p.PrincipalId,
                                                                                               p => p.RoleDefinitions)))
                    {

                        var t = listItem["PermissionRead"];
                      
                    }
```

**Reply by mvcsharepointdev** [reply]
@jansenbe thanks for the reply.. but what is i need to only load certain custom columns to improve performance ? is this possible and how?

also if i load all properties as you showed will the documents content also get loaded?

Thanks in advance for your help

**Reply by jansenbe** [reply]
@mvcsharepointdev : you today can't select individual properties using this approach, if you want that then you should use the Option C or Option B described in https://pnp.github.io/pnpcore/using-the-sdk/listitems-intro.html. Document content is never fetched automatically, you'll need to download document per document as described in https://pnp.github.io/pnpcore/using-the-sdk/files-intro.html#downloading-files

**Reply by mvcsharepointdev** [reply]
@jansenbe thanks again for your help. i think i am already using appraoch B "Getting list items via paging (no item limit)".. am i correct?

Second question, i went through the documentations for loading data, but i am still confused on how i can load specific custom columns? as inside the documentation it only shows how to load all properties or only the built-in properties/columns,, any advice? and sample codes please?

Thanks for your continuous help.

**Reply by jansenbe** [reply]
@mvcsharepointdev : Sorry, meant approach D :)

See the earlier link: A and B load custom columns but lack some of the "system" columns. If you want full control on the columns and/or apply a more complex filter then option C and D are better. Each have their own pro's and con's, you'll need to evaluate what works best for your scenario
